explorer/interpreter/type_checker.cpp

```diff
@@ -4455,8 +4455,9 @@ auto TypeChecker::ExpectReturnOnAllPaths(
 auto TypeChecker::DeclareCallableDeclaration(Nonnull<CallableDeclaration*> f,
                                              const ScopeInfo& scope_info)
     -> ErrorOr<Success> {
-  if (trace_stream_) {
-    **trace_stream_ << "** declaring function " << f->name() << "\n";
+  auto name = GetName(*f);
+  if (trace_stream_ && name) {
+    **trace_stream_ << "** declaring function " << *name << "\n";
```

Comment:
We would like want to keep those. Can we use `GetName` here as well?

Same applies to the other ones below

Reply:
Likewise as in my other comment, `f` is a `CallableDeclaration` which can either be a `FunctionDeclaration` or `DestructorDeclaration`, only the former of which still has the `name` getter now. I believe, then, that `GetName` is necessary here, no?

Reply:
Right, we need to log into the stream either way, so maybe we can add a `CARBON_CHECK` here as well, just to make sure we have a name, and keep the original `if (trace_stream_) {` condition, now that we know for certain that `name`/`GetName` can be unwrapped.

Reply:
Ah, I see. We want to keep the output. Got it! 🫡